### PR TITLE
Version 3.5.0

### PR DESCRIPTION
**Pull request by kasparsd**
Diff: https://github.com/xwp/stream/compare/tags/3.4.3..release/next

Notable bugfixes and features included in this realease:

- All the inline documentation updates and unit tests by @kidunot89 🚀
- #1082 Stream log search performance improvements
- #1114 readme copy update
- #1127 Stream record timestamp fix

Release Changelog:

- Inline PHP documentation improvements, props @kidunot89.
- WordPress coding standard fixes, props @kidunot89.
- TODO: bugfixes.

## Release Checklist

- [x] This pull request is to the `master` branch.
- [x] Release version follows [semantic versioning](https://semver.org). Does it include breaking changes?
- [x] Update changelog in `readme.txt`.
- [x] Bump version in `stream.php`.
- [x] Bump `Stable tag` in `readme.txt`.
- [x] Bump version in `classes/class-plugin.php`.
- [x] Draft a release [on GitHub](https://github.com/xwp/stream/releases/new).

Change `[ ]` to `[x]` to mark the items as done.

### CODE REVIEW

File: classes/class-author.php [hunk 1]
```diff
@@ -35,31 +50,38 @@ public function __construct( $user_id, $user_meta = array() ) {
...
-			throw new \Exception( "Unrecognized magic '$name'" );
+		switch ( $name ) {
+			case 'display_name':
+			case 'avatar_img':
+			case 'avatar_src':
+			case 'role':
+			case 'agent':
+				$getter = "get_{$name}";
+				return $this->$getter();
+			default:
```
Comment (kasparsd) [inline classes/class-author.php:70]: Noting this code change.

File: ui/js/admin.js [hunk 2]
```diff
@@ -68,6 +68,7 @@ jQuery(
 						width: '165px',
 						ajax: {
 							url: ajaxurl,
+							delay: 500,
```
Comment (kasparsd) [inline ui/js/admin.js:71]: Bugfix and UX improvement.

File: tests/tests/test-class-admin.php [hunk 3]
```diff
@@ -283,7 +283,7 @@ public function test_purge_scheduled_action() {
 
 		// Create (two day old) dummy records
 		$stream_data = $this->dummy_stream_data();
-		$stream_data['created'] = date( 'Y-m-d h:i:s', strtotime( '2 days ago' ) );
+		$stream_data['created'] = gmdate( 'Y-m-d h:i:s', strtotime( '2 days ago' ) );
```
Comment (kasparsd) [inline tests/tests/test-class-admin.php:286]: Note: WPCS fix which is actual code change.

File: connectors/class-connector-blogs.php [hunk 4]
```diff
@@ -74,7 +83,7 @@ public function get_context_labels() {
 
 			foreach ( $blogs as $blog ) {
 				$blog_details   = get_blog_details( $blog->blog_id );
-				$key            = sanitize_key( $blog_details->blogname );
+				$key            = sprintf( 'blog-%d', $blog->blog_id );
```
Comment (kasparsd) [inline connectors/class-connector-blogs.php:86]: Note: bugfix.

**Comment by kasparsd** [comment]
@kidunot89 Could you please smoke test this branch locally -- run linter and unit tests, use WP dev environment and confirm that logs are being populated as expected?

File: classes/class-author.php (hunk 1 above)
Comment (kidunot89) [reply classes/class-author.php:70]: I'm still not sure if this is tested or not.

**Pull request merged**